finetuner/tuner/paddle/__init__.py

```diff
@@ -232,3 +232,45 @@ def get_device(device: str):
         raise ValueError(
             f'Device {device} not recognized, only "cuda" and "cpu" are accepted'
         )
+
+
+class _ProjectionHead(nn.Layer):
+    """Projection head used internally for self-supervised training.
+    It is (by default) a simple 3-layer MLP to be attached on top of embedding model only for training purpose.
+    After training, it should be cut-out from the embedding model.
+    """
+
+    EPSILON = 1e-5
+
+    def __init__(self, in_features: int, output_dim: int = 128, num_layers: int = 3):
+        super().__init__()
+        self.head_layers = nn.LayerList()
+        is_last_layer = False
+        for idx in range(num_layers):
+            if idx == num_layers - 1:
+                is_last_layer = True
+            if not is_last_layer:
+                self.head_layers.append(
+                    nn.Linear(
+                        in_features=in_features,
+                        out_features=in_features,
```

Comment:
Why not to let the user specify the architecture? It seems like a big assumptiuon that all the layers are `in_feat, in_feat, in_feat` until the last layer

Reply:
i think there is not need, since this simple MLP trick works to improve SSL results. everyone is using Dense + BN + RELU combinations 2 - 3 layers, and it's a hidden head `_ProjectionHead` user will not notice

Reply:
I think that the interface would be as simple and the assumptions are way less.

I feel we are doing a lot of assumptions here

Reply:
do you mean that we fix number of layers and output dimension? @JoanFM

Reply:
u assume that the MLP is in_features x  num_layers x output_dim.

It seems weird that dimensionality is reduced slowly as layers are added

Reply:
i understand, it's tricky because we don't know the output dimension of the embedding model. We could "smooth", but my general impression is since it's a small MLP, it's not necessary